taxi/tests/test_views.py

```diff
@@ -0,0 +1,42 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(MANUFACTURER_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self):
+        Manufacturer.objects.create(name="BMW", country="Germany")
+        Manufacturer.objects.create(name="Tesla", country="USA")
+        self.user = get_user_model().objects.create_user(
+            username="test",
+            password="test123",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturers(self):
+        res = self.client.get(MANUFACTURER_URL)
+        self.assertEqual(res.status_code, 200)
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(res.context["manufacturer_list"]),
+            list(manufacturers),
+        )
+        self.assertTemplateUsed(res, "taxi/manufacturer_list.html")
```

Comment:
Ensure that the template `taxi/manufacturer_list.html` is correctly set up and used in the view for rendering the manufacturer list.